CONTRIBUTING.md

```diff
@@ -1,7 +1,186 @@
+
+#### Draft PRs
+If a PR is not yet ready for review, mark it as **Draft**. Once it is ready for review, mark it **Ready for Review**.
+
+#### Merging
+Perform merges to bring your feature branch up-to-date with master locally (do not issue a PR). This is very easy to do.
+1. Ensure you are on your feature branch (run `git branch` to find out; `git switch my-feature-branch` to switch to your feature branch).
+2. Make sure you have the latest changes (run `git fetch --all`)
+3. Perform the merge (run `git merge origin/master`, assuming that you want to merge the branch `master` from remote `origin` into `your-feature-branch`).
+
+#### Addressing reviews
+To address feedback from code review, implement changes and push to your feature branch. If you are certain that a reviewer's concern has been addressed by your new changes, hit the `Resolve conversation` button. If you are unsure, ask for follow up in the conversation and let the reviewer determine whether the feedback was addressed accordingly.
+
+### Code style and formatting
+The Lingua Franca compiler is implemented in Java and Kotlin. The overarching advice is to use each language's most widely used idioms and conventions, which are fortunately very similar. The code base is shipped with a [Spotless](https://github.com/diffplug/spotless) configuration to check and enforce style compliance. Lingua Franca code (e.g., tests) in this repository is also automatically formatted via Spotless.
+
+- To check that modified files are formatted correctly, run:
+```
+./gradlew spotlessCheck
+```
+- To apply the changes recommended by the formatter, run:
+```
+./gradlew spotlessApply
+```
+
+#### General guidelines
+- _Do not copy-paste code._ If you want to reuse code, factor it out into a method and call it.
+- _Keep methods concise._ As a rule of thumb, a method should fit on your screen so that it can be read without scrolling. We impose no hard limit on method length, but anything above 40 lines should be considered for breaking up.
+- _Keep classes concise._ Classes should be limited in scope and not become too large. Anything above 1000 lines probably needs refactoring.
+- _Do not leave FIXMEs_. Unaddressed `FIXME`s should not be allowed to pass code review unless they are accompanied with a link to a GitHub issue.
+
+_Comments_
+
+Please adhere to the following principles when writing documentation for your code:
+- Write descriptions in English.
+- Do not use contractions like "aren't" or "isn't".
+- It is OK to use phrases instead of complete sentences, in the interests of brevity. This holds especially in `@param` JavaDoc tag descriptions.
+- Use 3rd person (descriptive) not 2nd person (prescriptive), i.e., write "Compute the shortest path," not "Computes the shortest path."
```

Comment:
I am not very fit in grammar, but isn't "Computes ... " 3rd person?

Reply:
Jep, that's why the word *not* is written in before the quoted text :-)

Reply:
I see what you mean now. The description (which was a remainder of a previous draft) was copied without scrutiny :-) I fixed it. Thanks!